src/implementation/Client/HTTPClient/HTTPClient.ts

```diff
@@ -203,18 +203,18 @@ export default class HTTPClient implements IClient {
     if (res.status >= 200 && res.status <= 399) {
       return txtParsed;
     }
-    // 400 = Bad Request, 401 = Unauthorized, 404 = Not Found
-    else if (res.status === 400 || res.status === 401 || res.status === 404) {
-      throw new Error(JSON.stringify(txtParsed));
-    }
-    // 403 = Forbidden
-    else if (res.status === 403) {
-      throw new Error(JSON.stringify(txtParsed));
-    }
-    // 500 = Internal Server Error, 502 = Bad Gateway
-    else if (res.status === 500 || res.status === 502) {
-      throw new Error(JSON.stringify(txtParsed));
+    // 400 = Bad Request, 401 = Unauthorized, 404 = Not Found , 403 = Forbidden , 500 = Internal Server Error, 502 = Bad Gateway
+    else if (
```

Comment:
Can we change this to an indexOf check?

Reply:
sorry @XavierGeerinck  i didn't get it .can you provide me an example

Reply:
you want me to  create an array something like this 
const errorStatus  = [400, 401,403,500,502];

Reply:
Instead of capturing all error codes, why don't we handle anything outside [200,400) as error?

Reply:
@shubham1172  can you check now !